OFF-EP 0005b (#34)
openforcefield/standards: docs/standards/smirnoff.md

Electrostatic interactions are specified via the `<Electrostatics>` tag.
```XML
<Electrostatics version="0.4" periodic_potential="Ewald3D-ConductingBoundary" nonperiodic_potential="Coulomb" exception_potential="Coulomb" scale12="0.0" scale13="0.0" scale14="0.833333" scale15="1.0"/>
```

Some methods for computing electrostatic interactions are not valid for periodic systems, so
separate methods must be specified for periodic (`periodic_potential`) and non-periodic
(`nonperiodic_potential`) systems.

The `periodic_potential` attribute specifies the manner in which electrostatic interactions are to be computed in periodic systems. Allowed values are:

* `Ewald3D-ConductingBoundary` (default) denotes that a method like [particle mesh Ewald](https://docs.openmm.org/latest/userguide/theory.html#coulomb-interaction-with-particle-mesh-ewald) should be used

> * `Coulomb` denotes that the standard Coulomb potential should be used with specified cutoff `cutoff` and optionally switch width `switch_width`

Reviewer: ```suggestion
* `Coulomb` denotes that the standard Coulomb potential should be used with no cutoff or reaction-field attenuation
```

Reviewer 2: Is this right? I'm under the impression that "coulomb with no cutoff" is not possible for _periodic_ systems, since there are an infinite number of periodic particles to interact with. So I guess I'm assuming that "coulomb" here means "coulomb with a cutoff"

Reviewer 3: I agree that Coulomb with no cutoff (in periodic systems) is not typically something someone would want, and if you WERE going to do as Simon suggests you would probably want to SAY that that's the convention being used.

I think I'd (somewhat) advocate for taking Coulomb out entirely for periodic systems, but I'm also OK with amending this by specifying Simon's resolution (calculate only the nearest interaction).

Reviewer 3: (Once this issue is resolved I'm ready to see this PR merged.)

Reviewer: +1 for taking the Coulomb option out entirely for periodic systems

Reviewer 2: Ok, I've taken out the coulomb option out entirely for periodic systems

* A function denotes that the specified function should be used, which may make use of `cutoff`, `switch_width`, and/or `solvent_dielectric` terms

The `nonperiodic_potential` attribute specifies the manner in which electrostatic interactions are to be computed in non-periodic systems. Allowed values are:

* `Coulomb` (default) denotes that the standard Coulomb potential should be used with no cutoff or reaction-field attenuation
* A function denotes that the specified function should be used, which may make use of `cutoff`, `switch_width`, and/or `solvent_dielectric` terms

The `exception_potential` attribute specifies the treatment of intramolecular electrostatics exceptions, such as scaled 1-4 interactions. Allowed values are:

> * `Coulomb` (default) denotes that the standard Coulomb potential should be used

Reviewer: ```suggestion
* `Coulomb` (default) denotes that the standard Coulomb potential should be used with no cutoff or reaction-field attenuation
```

Currently, no child tags are used because the charge model is specified via different means (currently library charges or BCCs).

> For potentials where the cutoff determines the potential energy of the system (such as reaction field methods and `Coulomb`), the appropriate `cutoff` distance must also be specified, and the appropriate `switch_width` should be set to a numerical value if a switching function is to be used.

Reviewer: ```suggestion
For potentials where the cutoff determines the potential energy of the system (such as custom expression or reaction field methods), the appropriate `cutoff` distance must also be specified, and the appropriate `switch_width` should be set to a numerical value if a switching function is to be used.
```
